### PR TITLE
Add shield bar to force projector

### PR DESCRIPTION
**Pull request by DeltaNedas**
Implement https://github.com/Anuken/Mindustry-Suggestions/issues/1015

### CODE REVIEW

**Comment by Anuken** [comment]
image

**Comment by DeltaNedas** [comment]
oh god did the bundle change oh no

**Comment by Anuken** [comment]
you still can't use Pal.shield for this, it's transparent and looks awful

**Comment by DeltaNedas** [comment]
would Color.sky do for a blue color instead or should i make a Pal.shield thats not transparent

**Comment by Anuken** [comment]
sky is even worse, use something like accent

File: core/src/mindustry/world/blocks/defense/ForceProjector.java [hunk 1]
```diff
@@ -68,6 +69,12 @@ public void setStats(){
         stats.add(Stat.boostEffect, phaseShieldBoost, StatUnit.shieldHealth);
     }
 
+    @Override
+    public void setBars(){
+        super.setBars();
+        bars.add("shield", entity -> new Bar("stat.shieldhealth", Pal.accent, () -> 1 - ((ForceBuild) entity).buildup / breakage).blink(Color.white));
```
Comment (Anuken) [inline core/src/mindustry/world/blocks/defense/ForceProjector.java:75]: don't cast here, use `(ForceBuild entity)` as a lambda parameter

**Comment by DeltaNedas** [comment]
nah anuke you were wrong, casting is required

**Comment by Anuken** [comment]
> you were wrong

image

I said "use `(ForceBuild entity)` as a lambda parameter", which you didn't do. You just dropped the cast, which wouldn't compile for obvious reasons.

**Comment by DeltaNedas** [comment]
oh i thought it was a fixed Building thing h

**Comment by DeltaNedas** [comment]
gonna squash this too many h commits

**Comment by Anuken** [comment]
Now the bundle entry and color is messed up again. I'll just do this myself.

**Pull request closed without merging**